crates/system-services/src/distro.rs

```
@@ -0,0 +1,284 @@
+    let registry_config = registry_distro::RegistryConfig {
+        expired_interval: config.expired_period_sec,
+        renew_interval: config.renew_period_sec,
+        replicate_interval: config.replicate_period_sec,
+    };
+    let spell_distro = registry_distro::registry_spell(registry_config);
+    let mut trigger_config = TriggerConfig::default();
+    trigger_config.clock.start_sec = 1;
+    trigger_config.clock.period_sec = config.registry_period_sec;
+    let spell_distro = SpellDistro {
+        name: "registry-spell".to_string(),
+        air: spell_distro.air,
+        kv: spell_distro.init_data,
+        trigger_config,
+    };
+    let package = PackageDistro {
+        name: Decider.to_string(),
+        version: registry_distro::VERSION,
+        services: vec![service_distro],
+        spells: vec![spell_distro],
+        init: None,
+    };
+    Ok(package)
+}
+
+pub fn default_decider_distro(
+    decider_config: &DeciderConfig,
+    connector_config: &ConnectorConfig,
+) -> eyre::Result<PackageDistro> {
+    // prepare connector
+    let connector_service_distro = decider_distro::connector_service_modules();
+    let mut marine_config: TomlMarineConfig = toml::from_slice(connector_service_distro.config)?;
+    apply_binary_path_override(
+        &mut marine_config,
+        "curl_adapter",
+        "curl",
+        connector_config.curl_binary_path.clone(),
+    );
+
+    let service_distro = ServiceDistro {
+        modules: connector_service_distro.modules,
+        config: marine_config,
+        name: connector_service_distro.name.to_string(),
+    };
+
+    let wallet_key = match decider_config.wallet_key.clone() {
+        // TODO: set default wallet key somewhere in nox-distro, etc
+        //None => return Err(eyre!("Decider enabled, but wallet_key is not set. Please set it via env FLUENCE_ENV_CONNECTOR_WALLET_KEY or in Config.toml")),
+        None => "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f".to_string(),
```

I think throwing error makes more sense. Because `decider` won't work with that key anyway, and then Nox Admin will have hard time learning about that.

So fail early would be better than "fail somewhere deep after deploying someone else's deals".

---

@justprosh you had some problems with this place, can you comment to resolve what to do with it?

---

We should carefully disable decider for all non-relevant e2e tests and provide wallet addr where we need it. Also we should disable decider deployment by default, otherwise you just can't start nox container (also we have nothing about this in docs or readme)

---

> otherwise you just can't start nox container (also we have nothing about this in docs or readme)

Maybe we should provide a CLI flag to pass that private key. So that it's documented in `--help`, at least.

And maybe also a CLI flag to enable/disable system services. The ENV-based approach is too cumbersome to be the only one, I think.

> we should disable decider deployment by default

I'm not sure about that. Being able to deploy deals is crucial to Nox. It's just that now Nox requires wallet key and chain connection to work.

---

Yeah, required flag sounds suitable